extensions/amp-story/1.0/amp-story-page-attachment.js

```diff
@@ -339,14 +339,15 @@ export class AmpStoryPageAttachment extends DraggableDrawer {
         .querySelector('amp-story-page-outlink')
         ?.querySelector('a');
 
-      const pageAttachmentChild = this.element.parentElement
-        ?.querySelector('.i-amphtml-story-page-open-attachment-host')
-        .shadowRoot.querySelector('a.i-amphtml-story-page-open-attachment');
-
       if (pageOutlinkChild) {
         pageOutlinkChild.click();
-      } else if (pageAttachmentChild) {
-        triggerClickFromLightDom(pageAttachmentChild, this.element);
+      } else {
+        const pageAttachmentChild = this.element.parentElement
```

Comment:
const pageAttachmentChild = this.element.parentElement
          ?.querySelector('.i-amphtml-story-page-open-attachment-host')
          .shadowRoot.querySelector('a.i-amphtml-story-page-open-attachment');
          
          .i-amphtml-story-page-open-attachment-host' returns null and is only used when there is no page-outlink, it is a only refactor (no new code) to avoid this error.

Reply:
Sorry I still don't understand. What is the error? Why does your code change the behavior?

Reply:
If the refactor is not directly related to this PR I suggest reverting it and putting it in a separate PR.
This will help make the review process faster. 
It also keeps PRs focused to their concepts.

Reply:
It is directly related to the PR - in simple terms, it was avoiding a null pointer exception as this code was referencing
`.i-amphtml-story-page-open-attachment-host` even though it didn't exist. Our logic does not use that anyway if we are using page-outlink, so I moved it down.

After adding the other logic for `renderOpenAttachmentUI_`, the null pointer doesn't really happen anymore.

I could change it back, but that would be adding an unnecessary call in our code execution path for the `amp-story-page-outlink` layer - so I would consider this an optimization, avoiding an unnecessary call to something we don't use.

Reply:
If your new code no longer throws the error, I agree with @processprocess that we should move this change to a separate PR.

Reply:
Actually, I just double checked and it does throw an error - 
the page outlink will not work and will throw a null pointer due to the missing element (the null pointer triggers when you do the swipe up gesture), so the refactor is necessary.
I don't think it's a good idea to push non-working code to production. I'm going to add it back in.